### DISCUSSION TITLE
Fetch is caching 404 and returning 200 on subsequent requests

### DISCUSSION THREAD

**Discussion by st11x** [vercel/next.js#51323] [Help] [answered]
### Summary

Hi, I made a simple API endpoint app/api/hello/route.ts with this code to fetch from a known 404 page:

```typescript
import { NextResponse } from "next/server";

export async function GET(req: Request) {
    const res = await fetch("https://optinmonster.com/404");
    console.log("Res is", res.status);
    return new NextResponse("done", { status: res.status });
}
```

Hitting the endpoint the first time, I get a 404 response which is expected, but subsequent calls return a 200 response. Why is this so?

If I add the { cache: "no-store"} option to fetch, then it works as expected.  This is NextJS 13.2.4.

thanks
   Matt

**Comment by market-south** [comment]
Hi, 

As per the docs https://nextjs.org/docs/app/building-your-application/data-fetching/caching, all fetch() requests are cached by default. If the endpoint response is updated quite often, using `{ cache: "no-store"}` is a better option or else you can use `revalidate` to control the frequency of requests.

**Reply by st11x** [reply]
Thanks. I am aware that fetch requests get cached at the server side, but when the result is a 404, NextJS should return the "cached" 404. Now it is inventing responses until the next revalidate which seems wrong.

**Comment by MKraust** [comment]
I have a similar case. For example, API endpoint can return an article. I've set `revalidate: 10`.
Article can be published or not. When it's published we get 200 response with an article data. When it's unpublished we get 404.

So when I publish an article and visit it's page, then API response is cached. If I unpublish an article, next revalidation request would fail because of 404 status and I would see cached version of a page. BUT in this case I need to see 404.

I think we might need some setting which would allow us to control cache in case of errors while revalidating. According to [this section of docs](https://nextjs.org/docs/app/building-your-application/data-fetching/revalidating#error-handling-and-revalidation), we don't have such option now.

**Comment by fab1an** [comment, marked as answer]
This is a bug: https://github.com/vercel/next.js/issues/43831
